server-timing/defaults.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Server-Timing API default metrics
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Registers the default Server-Timing metrics for before rendering the template.
+ *
+ * These metrics should be registered as soon as possible.
+ *
+ * @since n.e.x.t
+ */
+function perflab_register_default_server_timing_before_template_metrics() {
+	$calculate_before_template_metrics = function() {
+		// WordPress execution prior to serving the template.
+		perflab_server_timing_register_metric(
+			'before-template',
+			array(
+				'measure_callback' => function( $metric ) {
+					// The 'timestart' global is set right at the beginning of WordPress execution.
+					$metric->set_value( ( microtime( true ) - $GLOBALS['timestart'] ) * 1000.0 );
+				},
+				'access_cap'       => 'exist',
+			)
+		);
+
+		if ( defined( 'SAVEQUERIES' ) && SAVEQUERIES ) {
+			// WordPress database query time before template.
+			perflab_server_timing_register_metric(
+				'before-template-db-queries',
+				array(
+					'measure_callback' => function( $metric ) {
+						// Store this value in a global to later subtract it from total query time after template.
+						$GLOBALS['perflab_query_time_before_template'] = array_reduce(
+							$GLOBALS['wpdb']->queries,
+							function( $acc, $query ) {
+								return $acc + $query[1];
+							},
+							0.0
+						);
```

Comment:
Got the following error on two hosting providers - Seravo and WPEngine. WPEngine has fixed the issue by itself and i don't see it anymore. Seravo still throws this error even with `SAVEQUERIES` enabled.

```
Fatal error:  Uncaught TypeError: array_reduce(): Argument #1 ($array) must be of type array, null given in /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/load.php:151
Stack trace:
#0 /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/load.php(151): array_reduce(NULL, Object(Closure), 0)
#1 /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/class-perflab-server-timing.php(98): {closure}(Object(Perflab_Server_Timing_Metric))
#2 /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/load.php(50): Perflab_Server_Timing->register_metric('template-db-que...', Array)
#3 /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/load.php(155): perflab_server_timing_register_metric('template-db-que...', Array)
#4 /data/wordpress/htdocs/wordpress/wp-includes/class-wp-hook.php(308): {closure}('')
#5 /data/wordpress/htdocs/wordpress/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#6 /data/wordpress/htdocs/wordpress/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#7 /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/class-perflab-server-timing.php(125): do_action('perflab_server_...')
#8 /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/class-perflab-server-timing.php(210): Perflab_Server_Timing->send_header()
#9 /data/wordpress/htdocs/wordpress/wp-includes/class-wp-hook.php(308): Perflab_Server_Timing->{closure}('')
#10 /data/wordpress/htdocs/wordpress/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#11 /data/wordpress/htdocs/wordpress/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#12 /data/wordpress/htdocs/wordpress/wp-includes/load.php(1124): do_action('shutdown')
#13 [internal function]: shutdown_action_hook()
#14 {main}
  thrown in /data/wordpress/htdocs/wp-content/plugins/performance-add-551-server-timing-api/server-timing/load.php on line 151
```

Reply:
The funniest part is that the server-timing header works correctly on another site that we have on Seravo hosting. 🤔

Reply:
@eugene-manuilov So for some reason `$GLOBALS['wpdb']->queries` is not an array in that situation? That definitely seems like a bug on their end, but happy to add a `isset && is_array` check here just to cater for weird setups :)

Reply:
Addressed in https://github.com/WordPress/performance/pull/553/commits/94304921e4e48af89f5961470a9c8f8659a44b36.

Reply:
> So for some reason `$GLOBALS['wpdb']->queries` is not an array in that situation?

Yep. That's weird but for some reason that's `NULL`. 🤷 

> Addressed in [9430492](https://github.com/WordPress/performance/commit/94304921e4e48af89f5961470a9c8f8659a44b36).

Thanks!